### PR TITLE
Gen0 segment should always be reported as Gen0 for DacHeapWalker

### LINKED ISSUE #96568: DAC fails to enumerate heap objects on .NET 7+ due to GC Regions
### Description

During heap object walking via `ICorDebugHeapEnum::Next`, `DacHeapWalker::MoveToNextObject()` fails to properly get the current segment / generation, causing it to skip calling `GetSegmentAndAllocRange()` for gen0 objects, so if it ends up in a dead area, it won't get itself back out and will fail to return objects to the enumerator.

SOS says the object under inspection is gen0, but when stopped on the access violation from `DacHeapWalker::GetSize` the check `isGen0` in `MoveToNextObject` claims it's gen2, causing it not to call `GetSegmentAndAllocRange()`
image

### Reproduction Steps

Known repros on clr versions 7.0.1423.51910, 8.0.23.53103. Setting DOTNET_GCRegionSize to a small value should increase the likelihood that even a template app can trigger the issue.

### Expected behavior

Objects are returned via the enumerator.

### Actual behavior

Enumerator gets stuck, and no further progress can be made.

### Regression?

Seems to be only the latest versions of the runtime, heap walking has not had this issue before.

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

DacHeapWalker::MoveToNextObject()

### PR DESCRIPTION
**Pull request by cshung**
Fixes #96568, the bug was most likely just a copy and paste error.

### CODE REVIEW

**Comment by hoyosjs** [comment]
I think there was some context around this. Let me see if my hunch is correct.

**Review by hoyosjs — APPROVED** [review]

File: src/coreclr/debug/daccess/request_svr.cpp [hunk 1]
```diff
@@ -400,14 +400,13 @@ HRESULT DacHeapWalker::InitHeapDataSvr(HeapData *&pHeaps, size_t &pCount)
...
-                    pHeaps[i].Segments[j].Generation = CorDebug_Gen0;
                 }
                 else
                 {
                     pHeaps[i].Segments[j].End = (CORDB_ADDRESS)seg->allocated;
-                    pHeaps[i].Segments[j].Generation = seg->flags & HEAP_SEGMENT_FLAGS_READONLY ? CorDebug_NonGC : CorDebug_Gen2;;
                 }
-
+                
+                pHeaps[i].Segments[j].Generation = CorDebug_Gen0;
```
Comment (Maoni0) [inline src/coreclr/debug/daccess/request_svr.cpp:409]: >pHeaps[i].Segments[j].Generation = CorDebug_Gen0; [](http://example.com/codeflow?start=16&length=49)

nit - perhaps move this to the top of this for loop to be consistent with other gens?

**Pull request merged**